test/distributed/test_ddp_with_rpc.py

```diff
@@ -0,0 +1,541 @@
+        self.name = name
+        self.ranks = ranks
+        self.group_name = group_name
+        self.group = None
+
+    def process_group(self) -> dist.ProcessGroup:
+        return self.group
+
+    def __enter__(self) -> dist.ProcessGroup:
+        gLogger.info(
+            f"Initing process group [{self.group_name}] by {self.name} with ranks {self.ranks}"
+        )
+        self.group = dist_c10d.new_group(ranks=self.ranks)
+        return self.group
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        gLogger.info(
+            f"Destroy process group [{self.group_name}] from process {self.name}."
+        )
+        if exc_type is not None:
+            raise exc_value
+        dist.destroy_process_group(self.group)
+
+
+class RemoteEM(nn.Module):
+    def __init__(self, num_embeddings: int, embedding_dim: int):
+        gLogger.info(f"Initing RemoteEM with {num_embeddings} {embedding_dim}")
+        super(RemoteEM, self).__init__()
+        init_em = [1] * embedding_dim
+        self.em = nn.EmbeddingBag(
+            num_embeddings,
+            embedding_dim,
+            _weight=torch.Tensor([init_em] * num_embeddings),
+        )
+
+    def forward(self, input: torch.Tensor):
+        gLogger.debug(f"Running RemoteEM.forward() on: {input}")
+        return self.em(input, offsets=torch.LongTensor(range(input.shape[0])))
+
+    def get_parameter_rrefs(self):
+        gLogger.debug(f"RemoteEM params: {list(self.parameters())}")
+        return [rpc.RRef(param) for param in self.parameters()]
+
+
+class RemoteNet(nn.Module):
+    def __init__(self, d_in: int, d_out: int):
+        gLogger.info(f"Initing RemoteNet with {d_in} {d_out}")
+        super(RemoteNet, self).__init__()
+        self.fc = nn.Linear(d_in, d_out)
+        self.relu = nn.ReLU()
+
+    def forward(self, input: torch.Tensor):
+        gLogger.debug(f"Running RemoteNet.forward() on: {input}")
+        return self.relu(self.fc(input))
+
+    def get_parameter_rrefs(self):
+        gLogger.debug(f"RemoteNet params: {list(self.parameters())}")
+        return [rpc.RRef(param) for param in self.parameters()]
+
+
+class HybridModel(nn.Module):
+    def __init__(
+        self,
+        remote_em_rref: rpc.RRef,
+        remote_net_rref: rpc.RRef,
+        process_group_for_ddp: dist.ProcessGroup = None,
+    ):
+        super(HybridModel, self).__init__()
+        self.remote_em_rref = remote_em_rref
+        self.remote_net_rref = remote_net_rref
+        self.remote_parameters_rrefs = _remote_method(
+            RemoteEM.get_parameter_rrefs, self.remote_em_rref
+        ) + _remote_method(RemoteNet.get_parameter_rrefs, self.remote_net_rref)
+        self.fc1 = nn.Linear(D_DENSE, D_DENSE)
+        self.fc2 = nn.Linear(D_HID, D_OUT)
+
+        if process_group_for_ddp is not None:
+            gLogger.info(f"Use DDP for the local nets.")
+            self.fc1 = DistributedDataParallel(
```

Comment:
nit: ddp_fc1

Reply:
Another data member `ddp_fc1` would cause branching (if/else) in `forward()` that I'd like to avoid.